pontoon/sync/tasks.py

```diff
@@ -53,18 +55,24 @@ def sync_project(self, project_pk, sync_log_pk, locale=None, no_pull=False, no_c
         start_time=now
     )
 
-    if not no_pull:
-        repos_changed = pull_changes(db_project)
-    else:
+    # We have to cache changed files before last revision is overriden by pull
+    if no_pull:
         repos_changed = True  # Assume changed.
+    else:
+        repos_changed, _ = pull_changes(db_project)
 
-    # If the repos haven't changed since the last sync and there are
+
+    # We have to find the changed files in the source repository
+    # If the wirepos haven't changed since the last sync and there are
     # no Pontoon-side changes for this project, quit early.
     if not force and not repos_changed and not db_project.needs_sync:
         log.info('Skipping project {0}, no changes detected.'.format(db_project.slug))
         project_sync_log.skip()
         return
 
+    if force:
+        db_project.repositories.update(last_synced_revisions='')
+
```

Comment:
I guess it was the easiest way to force scan of all files in the repo. Definetely i need to figure out more proper solution here.